lib/storage/sqlite/candidate-storage.js

```diff
@@ -60,6 +60,44 @@ class CandidateStorage {
       ))
       .catch((err) => cb(err));
   }
+
+  /**
+   * Fetch an array of Candidate objects by
+   * @param {candidateProcessor} callback
+   */
+  fetchAll (cb) {
+    var arrCandidates = [];
+    db.open(this.dbpath)
+      .then((db) => db.all('SELECT * FROM candidate_info'))
+      .then((rows) => {
+        var campaignPromises = [];
+
+        rows.forEach((row) => {
+          let candidate = new Candidate(row);
+          arrCandidates[row.candidate_id] = candidate;
+
+          // get campaigns
+          campaignPromises.push(db.all('SELECT * FROM campaign_info WHERE candidate_id = ?', row.candidate_id)
+          .then((campaigns) => campaigns));
+        });
+
+        return Promise.all(campaignPromises);
+      }).then((campaigns) => {
+        var campaignCandidateId;
+        for (let i = 0; i < campaigns.length; i++) {
+          campaignCandidateId = campaigns[i][0].candidate_id;
+          if (arrCandidates[campaignCandidateId]) {
+            arrCandidates[campaignCandidateId].campaigns = campaigns[i];
```

Comment:
The spec calls for the `campaigns` property of the `Candidate` object to be an array of  associated `Campaign` objects. This provides `CampaignInfo` objects. You need another query for each campaign in order to fetch and attach its associated `campaign_summary` array, and the `candidate_name` property attached to each campaign as well